sdk/core/Azure.Core/src/Pipeline/HttpPipeline.cs

```diff
@@ -19,6 +20,11 @@ public class HttpPipeline
 
         private readonly ReadOnlyMemory<HttpPipelinePolicy> _pipeline;
 
+        private readonly bool _internallyConstructed;
```

Comment:
Could you add comments to all these fields to explain what they are for. They are not very intuitive unless the person reading this code understands the per-request policy feature.

Reply:
Sure!